backend/util/util.go

```diff
@@ -127,19 +125,20 @@ func IsValidUsername(username string) bool {
 
 // IsAuthorized checks if a request contains the correct server key
 // Returns true if the provided key is equal to the evironment variable
-// Returns false otherwise
-func IsAuthorized(w http.ResponseWriter, r *http.Request) bool {
+// Returns false and error otherwise
+func IsAuthorized(w http.ResponseWriter, r *http.Request) (bool, error) {
 	decoder := json.NewDecoder(r.Body)
 	var req reqStruct
 
 	err := decoder.Decode(&req)
 	if err != nil {
-		// couldn't decode the POST data into the right
-		// JSON object
-		return false
+		// couldn't decode the POST data into JSON
+		return false, err
 	}
-	if req.Secret == os.Getenv("secretKey") {
-		return true
+
+	if req.Secret != os.Getenv("secretKey") {
+		fmt.Println(req.Secret)
+		return false, errors.New("Incorrect 'secret'")
```

Comment:
I think "Incorrect secretKey environvment variable" would be more descriptive if the user didn't know that a `secretKey` needed to set

Reply:
Didn't implement because I'm cooler.